scheduler/src/cook/mesos/rebalancer.clj

```diff
@@ -186,25 +211,39 @@
    value
    host->spare-resources A map from host to spare resources.
    user->dru-divisors A map from user to dru divisors."
-  [db running-task-ents pending-job-ents host->spare-resources]
-  (let [user->dru-divisors (dru/init-user->dru-divisors db running-task-ents pending-job-ents)
+  [db running-task-ents pending-job-ents host->spare-resources category]
+  (let [running-task-ents (filter (fn [task]
+                                    (-> task
+                                        :job/_instance
+                                        util/categorize-job
+                                        (= category)))
+                                  running-task-ents)
+        user->dru-divisors (dru/init-user->dru-divisors db running-task-ents pending-job-ents)
         user->sorted-running-task-ents (->> running-task-ents
                                             (group-by util/task-ent->user)
                                             (map (fn [[user task-ents]]
-                                                   [user (into (sorted-set-by (util/same-user-task-comparator-penalize-backfill)) task-ents)]))
+                                                   [user (into (sorted-set-by (case category
+                                                                                :normal (util/same-user-task-comparator-penalize-backfill)
+                                                                                :gpu (util/same-user-task-comparator))) task-ents)]))
                                             (into {}))
-        task->scored-task (into (pm/priority-map-keyfn (comp - :dru))
-                                (dru/sorted-task-scored-task-pairs user->sorted-running-task-ents user->dru-divisors))]
-    (->State task->scored-task user->sorted-running-task-ents host->spare-resources user->dru-divisors)))
+        task->scored-task (into (pm/priority-map-keyfn (case category
+                                                         :normal (comp - :dru)
+                                                         :gpu -))
```

Comment:
the `user->dru-divisors` map maps from usernames to `dru/ScoredTask`. It turns out that we don't need all the extra metadata for gpus, so we just map from usernames to DRU for GPUs.